### PR TITLE
Closes #67 - AppDatabase, AppRepository, Executors, and SampleData

### LINKED ISSUE #67: Updates database components and populates with SampleData
Update app to follow common architecture patterns and dependency injection. Repository and models updated. Correct error preventing named time zone usage, resulting in errant Daylight Saving Time behavior. Add sample data for proof of concept.

DOD: Database initialized with sample data for all entities, model(s) demonstrate correct functionality under Daylight Saving Time, sample data is inserted cleanly upon each execution.

### PR DESCRIPTION
**Pull request by gomezfox**

### CODE REVIEW

**Review by haythamodeh — APPROVED** [review]
looks good

File: app/src/main/java/edu/cascadia/mobas/campusguidebook/data/database/AppDatabase.java [hunk 1]
```diff
@@ -38,52 +47,36 @@
...
+                                    .addCallback(new RoomDatabase.Callback() {
+                                        @Override
+                                        public void onCreate(@NonNull SupportSQLiteDatabase sqLiteDatabase) {
+                                            super.onCreate(sqLiteDatabase);
+                                            Log.d("AppDatabase", "RoomDatabase.Callback Executed");
+                                        }
+                                    })
+                                    .build();  // end of RoomDatabaseBuilder
+                    // TODO: REPLACE THE FOLLOWING BEFORE RELEASE
+                    SampleData.addAll(sAppDatabase, appExecutors);
```
Comment (Bansenauer-Cascadia) [inline app/src/main/java/edu/cascadia/mobas/campusguidebook/data/database/AppDatabase.java:75]: Nicely done. Clear documentation

File: app/src/main/java/edu/cascadia/mobas/campusguidebook/data/repository/AppRepository.java [hunk 2]
```diff
@@ -47,290 +44,118 @@ public static AppRepository getInstance() {
...
-            // our Event model in dao.
-            Sustainabilitydao.delete(models[0]);
-            return null;
-        }
+    public void delete(User... users) {
+        mAppExecutors.diskIO().execute( () -> mAppDatabase.runInTransaction( () -> {
+            for (User user : users) {
+                mAppDatabase.UserDao().delete(user);
+            }
+        }));
```
Comment (Bansenauer-Cascadia) [inline app/src/main/java/edu/cascadia/mobas/campusguidebook/data/repository/AppRepository.java:159]: Excellent generalization of CRUD operations coded for more than one entity at a time.  Beautiful work

**Review by Bansenauer-Cascadia — COMMENTED** [review]
Great work. Since this is a major architectural update, I encourage everyone to pull this branch, test it out, and pull it into your working branch to see how behaves with your code. Then approve or request changes as soon as you can.

**Review by odjuliarso — APPROVED** [review]
approved.

**Review by isaakwheeler — APPROVED** [review]
looking good!

**Pull request merged**